cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -90,15 +91,16 @@ func NewKubeStaticPodPathManager(realDir, tempDir, backupDir, backupEtcdDir stri
 
 // NewKubeStaticPodPathManagerUsingTempDirs creates a new instance of KubeStaticPodPathManager with temporary directories backing it
 func NewKubeStaticPodPathManagerUsingTempDirs(realManifestDir string, saveManifestsDir, saveEtcdDir bool) (StaticPodPathManager, error) {
-	upgradedManifestsDir, err := constants.CreateTempDirForKubeadm("kubeadm-upgraded-manifests")
+	kubernetesDir := filepath.Dir(realManifestDir)
```

Comment:
Again, I don't like this one, you need a top-down approach for this one.
Please, sync up with @ereslibre as you are intersecting with one of his PRs on this part of the code.

Reply:
For completion, the PR that @rosti is referring to is: https://github.com/kubernetes/kubernetes/pull/76929